x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateFormat.java

```diff
@@ -57,12 +57,20 @@ protected TypeResolution resolveType() {
             return new TypeResolution("Unresolved children");
         }
 
-        TypeResolution resolution = isDate(field, sourceText(), FIRST);
+        TypeResolution resolution;
+        if (format != null) {
+            resolution = argumentTypesAreSwapped(format.dataType(), field.dataType(), DataTypes::isString, sourceText());
+            if (resolution.unresolved()) {
+                return resolution;
+            }
+        }
+
+        resolution = isDate(field, sourceText(), format == null ? FIRST : SECOND);
         if (resolution.unresolved()) {
             return resolution;
         }
         if (format != null) {
-            resolution = isStringAndExact(format, sourceText(), SECOND);
+            resolution = isStringAndExact(format, sourceText(), FIRST);
```

Comment:
nit: similarly, there's 3 checks for `format != null` here. IMHO this would be easier to read if there was just an if-else. (Also applies to a lesser degree below in `info()`).

Reply:
I think the `format != null` checks will have to stay, though, because the format string is optional, right?

Reply:
Never mind, here specifically the one branch in line 61 will go away and then an overarching if-else will likely be less readable again. Anyway, meant as a suuuuper tiny nit anyway :)